cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ArtifactHttpHandlerInternal.java

```diff
@@ -90,6 +101,44 @@ public void listArtifacts(HttpRequest request, HttpResponder responder,
     }
   }
 
+  @GET
+  @Path("/namespaces/{namespace-id}/artifacts/{artifact-name}/versions/{artifact-version}/download")
+  public void getArtifactBytes(HttpRequest request, HttpResponder responder,
```

Comment:
If we have the file fetcher endpoint, do we still need this? https://github.com/cdapio/cdap/pull/13380

Reply:
@wyzhang had some concerns, I'll let him elaborate

Reply:
This PR is already very large, I'll make this change in a follow up PR once #13380 is merged